fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnBE.java

```diff
@@ -216,43 +217,11 @@ private Map<String, Expression> evalOnBE(Map<String, Map<String, TExpr>> paramMa
-                        PrimitiveType primitiveType = PrimitiveType.fromThrift(Objects.requireNonNull(tPrimitiveType));
                         Expression ret;
                         if (e1.getValue().getSuccess()) {
-                            DataType type;
-                            if (PrimitiveType.ARRAY == primitiveType
-                                    || PrimitiveType.MAP == primitiveType
-                                    || PrimitiveType.STRUCT == primitiveType
-                                    || PrimitiveType.AGG_STATE == primitiveType) {
-                                ret = constMap.get(e1.getKey());
-                            } else {
-                                if (primitiveType == PrimitiveType.CHAR) {
-                                    Preconditions.checkState(pScalarType.hasLen(),
-                                            "be return char type without len");
-                                    type = CharType.createCharType(pScalarType.getLen());
-                                } else if (primitiveType == PrimitiveType.VARCHAR) {
-                                    Preconditions.checkState(pScalarType.hasLen(),
-                                            "be return varchar type without len");
-                                    type = VarcharType.createVarcharType(pScalarType.getLen());
-                                } else if (primitiveType == PrimitiveType.DECIMALV2) {
-                                    type = DecimalV2Type.createDecimalV2Type(
-                                            pScalarType.getPrecision(), pScalarType.getScale());
-                                } else if (primitiveType == PrimitiveType.DATETIMEV2) {
-                                    type = DateTimeV2Type.of(pScalarType.getScale());
-                                } else if (primitiveType == PrimitiveType.DECIMAL32
-                                        || primitiveType == PrimitiveType.DECIMAL64
-                                        || primitiveType == PrimitiveType.DECIMAL128
-                                        || primitiveType == PrimitiveType.DECIMAL256) {
-                                    type = DecimalV3Type.createDecimalV3TypeLooseCheck(
-                                            pScalarType.getPrecision(), pScalarType.getScale());
-                                } else {
-                                    type = DataType.fromCatalogType(ScalarType.createType(
-                                            PrimitiveType.fromThrift(tPrimitiveType)));
-                                }
-                                ret = Literal.of(e1.getValue().getContent()).castTo(type);
-                            }
+                            PTypeDesc pTypeDesc = e1.getValue().getPType();
+                            DataType type = convertToNereidsType(pTypeDesc.getTypesList(), 0).key();
+                            ret = Literal.of(e1.getValue().getContent()).castTo(type);
```

Comment:
i think, we'd better refactor fold constant on be. we should not rely on string as ser/de infra. we should use a better way to ser/de literal expression just like what we do in thrift.